scripts/gendocs/generate

```diff
@@ -142,21 +144,22 @@ $output
 EOF
 }
 
-genclireference "docs" "cmd/acmesolver" "acmesolver"
-genclireference "docs" "cmd/cainjector" "cainjector"
-genclireference "docs" "cmd/ctl" "cmctl"
-genclireference "docs" "cmd/controller" "controller"
-genclireference "docs" "cmd/webhook" "webhook"
-
 # The branches named here exist in the `jetstack/cert-manager` repo.
 
-genversion "release-1.7" "next-docs"
+genversionwithcli "release-1.7" "next-docs"
 
 # In cert-manager 1.6 cert-manager.io and acme.cert-manager.io alpha and beta
 # API versions are no longer served, but the apis are still in the codebase. We
 # don't want to show the docs for those API versions so this is a workaround.
 # This will only be necessary for the release-1.6 branch.
 checkout "release-1.6"
+
+genclireference "docs" "cmd/acmesolver" "acmesolver"
+genclireference "docs" "cmd/cainjector" "cainjector"
+genclireference "docs" "cmd/ctl" "cmctl"
+genclireference "docs" "cmd/controller" "controller"
+genclireference "docs" "cmd/webhook" "webhook"
```

Comment:
And if these lines are deliberately placed after the `checkout` command, then I'd add a comment explaining why it's important, so that someone doesn't come alone and rearrange this code.

Reply:
I have added a comment here covering this!